src/engine/controls/bpmcontrol.cpp

```diff
@@ -99,61 +123,61 @@ BpmControl::BpmControl(const QString& group,
     // bpm_down controls.
     // bpm_up / bpm_down steps by kBpmRangeStep
     // bpm_up_small / bpm_down_small steps by kBpmRangeSmallStep
-    m_pEngineBpm = new ControlLinPotmeter(
+    m_pEngineBpm = std::make_unique<ControlLinPotmeter>(
             ConfigKey(group, "bpm"),
             kBpmRangeMin,
             kBpmRangeMax,
             kBpmRangeStep,
             kBpmRangeSmallStep,
             true);
     m_pEngineBpm->set(0.0);
-    connect(m_pEngineBpm, &ControlObject::valueChanged,
-            this, &BpmControl::slotUpdateRateSlider,
+    connect(m_pEngineBpm.get(),
+            &ControlObject::valueChanged,
+            this,
+            &BpmControl::slotUpdateRateSlider,
             Qt::DirectConnection);
 
-    m_pButtonTap = new ControlPushButton(ConfigKey(group, "bpm_tap"));
-    connect(m_pButtonTap, &ControlObject::valueChanged,
-            this, &BpmControl::slotBpmTap,
+    // Tap the (file) BPM
+    m_pBpmTap = std::make_unique<ControlPushButton>(ConfigKey(group, "bpm_tap"));
+    connect(m_pBpmTap.get(),
+            &ControlObject::valueChanged,
+            this,
+            &BpmControl::slotBpmTap,
             Qt::DirectConnection);
-
-    m_pTranslateBeats = new ControlPushButton(ConfigKey(group, "beats_translate_curpos"));
-    connect(m_pTranslateBeats, &ControlObject::valueChanged,
-            this, &BpmControl::slotBeatsTranslate,
+    connect(&m_bpmTapFilter,
+            &TapFilter::tapped,
+            this,
+            &BpmControl::slotBpmTapFilter,
             Qt::DirectConnection);
 
-    m_pBeatsTranslateMatchAlignment = new ControlPushButton(ConfigKey(group, "beats_translate_match_alignment"));
-    connect(m_pBeatsTranslateMatchAlignment, &ControlObject::valueChanged,
-            this, &BpmControl::slotBeatsTranslateMatchAlignment,
+    // Tap the tempo (playback speed)
+    m_pTempoTap = std::make_unique<ControlPushButton>(ConfigKey(group, "tempo_tap"));
+    connect(m_pTempoTap.get(),
+            &ControlObject::valueChanged,
+            this,
+            &BpmControl::slotTempoTap,
             Qt::DirectConnection);
-
-    connect(&m_tapFilter, &TapFilter::tapped,
-            this, &BpmControl::slotTapFilter,
+    connect(&m_tempoTapFilter,
+            &TapFilter::tapped,
+            this,
+            &BpmControl::slotTempoTapFilter,
             Qt::DirectConnection);
 
-    m_pBpmLock = std::make_unique<ControlPushButton>(
-            ConfigKey(group, "bpmlock"), false);
-    m_pBpmLock->setButtonMode(ControlPushButton::TOGGLE);
-    m_pBpmLock->connectValueChangeRequest(
```

Comment:
because of this change m_pBpmLock is left null and crashes when used in trackBpmLockChanged